test/mktemp.js

```diff
@@ -0,0 +1,112 @@
+var shell = require('..'),
+    common = require('../src/common');
+
+var assert = require('assert'),
+    fs = require('fs'),
+    path = require('path');
+
+shell.config.silent = true;
+shell.rm('-rf', 'tmp');
+shell.mkdir('tmp');
+
+// Not testing chmod on Windows.
+var isNotWindows = common.platform !== 'win';
+var result;
+
+
+//
+// Invalids
+//
+
+shell.touch(path.resolve('.', 'tmp/tmp.A'));
+result = shell.mktemp(path.resolve('.', 'tmp/tmp.A')); // Guarenteed to have a confilct, no randomness.
```

Comment:
Is it possible to actually specify a name like this? I just tried `$ mktemp foo` and it says "too few X's in template 'foo'" (and exits with status 1).